src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProvider.java

```diff
@@ -332,12 +335,12 @@ public void deleteInfrastructure(Infrastructure infrastructure) {
     }
 
     @Override
-    public String addToInstancePublicIp(Infrastructure infrastructure, String instanceId) {
+    public String addToInstancePublicIp(Infrastructure infrastructure, String instanceId, String optionalDesiredIp) {
```

Comment:
Because, according to Intellij good practices, Optional are not intended to be used as parameters:
```
Reports any uses of java.util.Optional<T>, java.util.OptionalDouble, java.util.OptionalInt, java.util.OptionalLong or com.google.common.base.Optional as the type for a field or a parameter. Optional was designed to provide a limited mechanism for library method return types where there needed to be a clear way to represent "no result". Using a field with type java.util.Optional is also problematic if the class needs to be Serializable, which java.util.Optional is not.
```